core/ir/x86/decode.c

```diff
@@ -2410,6 +2436,22 @@ decode_get_tuple_type_input_size(const instr_info_t *info, decode_info_t *di)
         di->input_size = OPSZ_NA;
 }
 
+/* TODO i#6238: Not all opcodes have been reviewed.
+ * In case an opcode has not been reviewed,
+ * the default category assigned to it is DR_INSTR_CATEGORY_OTHER.
+ */
+static inline void
+decode_category(instr_t *instr)
+{
+    if (instr) {
+        if (op_instr[instr->opcode])
+            instr_set_category(instr, op_instr[instr->opcode]->category);
+        else
+            /* nonvalid opcode */
+            instr_set_category(instr, DR_INSTR_CATEGORY_UNCATEGORIZED);
+    }
+}
+
```

Comment:
Wow, the decode_table.c diff is too big and github won't even let me expand it or comment on it!  So adding comments here:

>+    {OP_add,  0x010000, DR_INSTR_CATEGORY_INT | DR_INSTR_CATEGORY_MATH, "add",  Ev, xx, Gv, Ev, xx, mrm, fW6, tfb[0x00]},
>+    {OP_add,  0x020000, DR_INSTR_CATEGORY_INT | DR_INSTR_CATEGORY_MATH, "add",  Gb, xx, Eb, Gb, xx, mrm, fW6, tfb[0x01]},

I was going to say that moving into the decode entries was good because here we have the same opcode being either a load or a store: but I guess for this case we plan to add _LOAD or _STORE in decode_category() so maybe these could have been in an opcode array.  But I bet we find other cases where it's better in the entries.

Given that we want to be able to view the table lines without them being incredibly wide, I think it's worth sacrificing a little verbosity by using abbreviations for the long enum names.  Maybe `s/DR_INSTR_CATEGORY_INT | DR_INSTR_CATEGORY_MATH/catInt|catMath/`?

Reply:
@derekbruening 
Do you suggest to replace enum to the following?
typedef enum {
    catUncategorized = 0x0, /**< Uncategorized. */
    catInt = 0x1,           /**< Integer arithmetic operations. */
    catFP = 0x2,            /**< Floating-Point operations. */
    catLoad = 0x4,          /**< Loads. */
    catStore = 0x8,         /**< Stores. */
    catBranch = 0x10,       /**< Branches. */
    catSIMD = 0x20,  /**< Operations with vector registers (SIMD). */
    catState = 0x40, /**< Saves, restores, or queries processor state. */
    catMove = 0x80,  /**< Moves value from one location to another. */
    catConvert = 0x100, /**< Converts to or from value. */
    catMath = 0x200, /**< Performs arithmetic or conditional operations. */
    catOther = 0x400 /**< Other types of instructions. */
} dr_instr_category_t;

Reply:
No, I would keep the enum how it is.  I would define local macros just in decode_table.c to shorten those long lines (like the existing short names for the various other fields).